src/jit/codegenxarch.cpp

```diff
@@ -4653,7 +4653,7 @@ void CodeGen::genStoreInd(GenTreePtr node)
         }
         else
         {
-            getEmitter()->emitInsMov(ins_Store(data->TypeGet()), emitTypeSize(storeInd), storeInd);
+            getEmitter()->emitInsMov(ins_Store(data->TypeGet()), emitActualTypeSize(storeInd->TypeGet()), storeInd);
```

Comment:
I think this change is incorrect: if we are issuing an indirect store, we need to issue it with the correct type. IIUC, `emitActualTypeSize` will widen small types.